x-pack/legacy/plugins/apm/public/components/shared/Links/apm/TransactionLink.tsx

```diff
@@ -7,6 +7,8 @@
 import React from 'react';
 import { Transaction } from '../../../../../typings/es_schemas/ui/Transaction';
 import { APMLink } from './APMLink';
+import { useUrlParams } from '../../../../hooks/useUrlParams';
+import { pickKeys } from '../../../../utils/pickKeys';
 
 interface TransactionLinkProps {
   transaction: Transaction | undefined;
```

Comment:
`TransactionLink` is a bit different than all the others because it takes a `Transaction` as argument. Did you consider changing that to individual params like:

```js
TransactionLink({ serviceName, traceId, transactionId, transactionName, transactionType }
```

Reply:
Yeah, I think flattening those is an improvement, will change 👍